generated/index.md

```diff
@@ -5661,14 +6811,6 @@
 * [2021-04-01](machinelearningservices/microsoft.machinelearningservices/2021-04-01/types.md#resource-microsoftmachinelearningservicesworkspacesservices2021-04-01)
 
 ## microsoft.maintenance
-### microsoft.maintenance/applyupdates
-* [2018-06-01-preview](maintenance/microsoft.maintenance/2018-06-01-preview/types.md#resource-microsoftmaintenanceapplyupdates2018-06-01-preview)
-* [2020-04-01](maintenance/microsoft.maintenance/2020-04-01/types.md#resource-microsoftmaintenanceapplyupdates2020-04-01)
-* [2020-07-01-preview](maintenance/microsoft.maintenance/2020-07-01-preview/types.md#resource-microsoftmaintenanceapplyupdates2020-07-01-preview)
-* [2021-04-01-preview](maintenance/microsoft.maintenance/2021-04-01-preview/types.md#resource-microsoftmaintenanceapplyupdates2021-04-01-preview)
-* [2021-05-01](maintenance/microsoft.maintenance/2021-05-01/types.md#resource-microsoftmaintenanceapplyupdates2021-05-01)
-* [2021-09-01-preview](maintenance/microsoft.maintenance/2021-09-01-preview/types.md#resource-microsoftmaintenanceapplyupdates2021-09-01-preview)
-
```

Comment:
The `applyUpdates` resource uses [a literal name segment on PUT](https://github.com/Azure/azure-rest-api-specs/blob/49fc16354df7211f8392c56884a3437138317d1f/specification/maintenance/resource-manager/Microsoft.Maintenance/stable/2020-04-01/Maintenance.json#L250-L312) and [a parameterized name segment on GET](https://github.com/Azure/azure-rest-api-specs/blob/49fc16354df7211f8392c56884a3437138317d1f/specification/maintenance/resource-manager/Microsoft.Maintenance/stable/2020-04-01/Maintenance.json#L103-L172). This leads to [the resource getting rejected in the type generator](https://github.com/Azure/bicep-types-az/blob/9127c6871bb3e0e9939a79ad9b77dad53c123d24/src/autorest.bicep/src/type-generator.ts#L108), but only when readonly resources are loaded. This is closely related to the issue that caused `Microsoft.Billing@2017-02-27-preview` to generate types files with no types (which I was hoping not to have to solve in this PR).